LinuxCleaner_42.sh

```diff
@@ -0,0 +1,172 @@
+fi
+
+function clean_glob {
+	# don't do anything if argument count is zero (unmatched glob).
+	if [ -z "$1" ]; then
+		return 0
+	fi
+
+	if [ $should_log -eq 1 ]; then
+		for arg in "$@"; do
+			du -sh "$arg" 2>/dev/null
+		done
+	fi
+
+	/bin/rm -rf "$@" &>/dev/null
+
+	return 0
+}
+
+function clean {
+	# to avoid printing empty lines
+	# or unnecessarily calling /bin/rm
+	# we resolve unmatched globs as empty strings.
+	shopt -s nullglob
+
+	echo -ne "\033[38;5;208m"
+
+	#42 Caches
+	clean_glob "$HOME"/.42*
+	clean_glob "$HOME"/.zcompdump*
+
+	#Trash
+	clean_glob "$HOME"/.local/share/Trash/files/*
+	clean_glob "$HOME"/.local/share/Trash/info/*
+
+	#General Cache files
+	clean_glob "$HOME"/.cache/*
+
+	#Package manager caches
+	clean_glob "$HOME"/.cache/pip/*
+	clean_glob "$HOME"/.npm/_cacache/*
+	clean_glob "$HOME"/.yarn/cache/*
+	clean_glob "$HOME"/.cargo/registry/cache/*
+	clean_glob "$HOME"/.gem/ruby/*/cache/*
+
+	#Browser Caches - Firefox
+	clean_glob "$HOME"/.mozilla/firefox/*/cache2/*
+	clean_glob "$HOME"/.cache/mozilla/firefox/*/cache2/*
+	clean_glob "$HOME"/.mozilla/firefox/*/startupCache/*
+	clean_glob "$HOME"/.mozilla/firefox/*/OfflineCache/*
+
+	#Browser Caches - Chrome/Chromium
+	clean_glob "$HOME"/.config/google-chrome/Default/Cache/*
+	clean_glob "$HOME"/.config/google-chrome/Profile*/Cache/*
+	clean_glob "$HOME"/.config/chromium/Default/Cache/*
+	clean_glob "$HOME"/.config/chromium/Profile*/Cache/*
+	clean_glob "$HOME"/.cache/google-chrome/Default/Cache/*
+	clean_glob "$HOME"/.cache/chromium/Default/Cache/*
+
+	#Browser Caches - Brave
+	clean_glob "$HOME"/.config/BraveSoftware/Brave-Browser/Default/Cache/*
+	clean_glob "$HOME"/.config/BraveSoftware/Brave-Browser/Profile*/Cache/*
+	clean_glob "$HOME"/.config/BraveSoftware/Brave-Browser/Crashpad/completed/*
+
+
+	#Application Caches
+	clean_glob "$HOME"/.config/Code/Cache/*
+	clean_glob "$HOME"/.config/Code/CachedData/*
+	clean_glob "$HOME"/.config/Code/User/workspaceStorage/*
+	clean_glob "$HOME"/.vscode/extensions/*/node_modules/*
+	clean_glob "$HOME"/.config/discord/Cache/*
+	clean_glob "$HOME"/.config/discord/Code\ Cache/*
+	clean_glob "$HOME"/.config/Slack/Cache/*
+	clean_glob "$HOME"/.config/Slack/Service\ Worker/CacheStorage/*
+	clean_glob "$HOME"/.config/spotify/Users/*/offline-cache/*
+	clean_glob "$HOME"/.config/spotify/PersistentCache/*
+    clean_glob "$HOME"/.config/*/Cache/*
+    clean_glob "$HOME"/.config/*/cache/*
+
+	#Thumbnails
+	clean_glob "$HOME"/.cache/thumbnails/*
+	clean_glob "$HOME"/.thumbnails/*
+
+	#Temporary files
+	clean_glob /tmp/*
+	clean_glob "$HOME"/.tmp/*
+	clean_glob "$HOME"/tmp/*
+
+	#Things related to pool (piscine)
+	clean_glob "$HOME"/Desktop/Piscine\ Rules\ *.mp4
+	clean_glob "$HOME"/Desktop/PLAY_ME.webloc
+
+    # Docker user caches
+    clean_glob "$HOME"/.docker/*/cache/*
+
+	#Recent files and history
+	# clean_glob "$HOME"/.local/share/recently-used.xbel*
+	# clean_glob "$HOME"/.bash_history
+	# clean_glob "$HOME"/.zsh_history
```

Comment:
You can remove these lines if uneeded, I believe we shouldn't clean the shell history